eslint.config.mjs

```diff
@@ -0,0 +1,423 @@
+      },
+
+      ecmaVersion: 5,
+      sourceType: 'script',
+    },
+
+    rules: {
+      'import/no-commonjs': 'off',
+    },
+  },
+  {
+    files: ['**/*.ts', '**/*.tsx'],
+
+    extends: [
+      js.configs.recommended,
+      tseslint.configs.strictTypeChecked,
+      tseslint.configs.stylisticTypeChecked,
+      react.configs.flat.recommended,
+      reactHooks.configs['recommended-latest'],
+      jsxA11Y.flatConfigs.recommended,
+      importPlugin.flatConfigs.recommended,
+      importPlugin.flatConfigs.typescript,
+      promisePlugin.configs['flat/recommended'],
+      jsdoc.configs['flat/recommended-typescript'],
+    ],
+
+    languageOptions: {
+      ecmaVersion: 5,
+      sourceType: 'script',
+
+      parserOptions: {
+        projectService: true,
+        tsconfigRootDir: import.meta.dirname,
+      },
+    },
+
+    rules: {
+      // Disable formatting rules that have been enabled in the base config
+      indent: 'off',
+
+      // This is not needed as we use noImplicitReturns, which handles this in addition to understanding types
+      'consistent-return': 'off',
+
+      'formatjs/enforce-plural-rules': 'off',
+
+      'import/consistent-type-specifier-style': ['error', 'prefer-top-level'],
+      'import/no-default-export': 'warn',
+
+      'jsdoc/require-jsdoc': 'off',
+
+      'react/prefer-stateless-function': 'warn',
+      'react/function-component-definition': [
+        'error',
+        {
+          namedComponents: 'arrow-function',
+        },
+      ],
+      'react/jsx-uses-react': 'off', // not needed with new JSX transform
+      'react/react-in-jsx-scope': 'off', // not needed with new JSX transform
+      'react/prop-types': 'off',
+
+      '@typescript-eslint/consistent-type-definitions': ['warn', 'interface'],
+      '@typescript-eslint/consistent-type-exports': 'error',
+      '@typescript-eslint/consistent-type-imports': 'error',
+      '@typescript-eslint/prefer-nullish-coalescing': [
+        'error',
+        {
+          ignorePrimitives: {
+            boolean: true,
+          },
+        },
+      ],
+      '@typescript-eslint/no-restricted-imports': [
+        'warn',
+        {
+          name: 'react-redux',
+          importNames: ['useSelector', 'useDispatch'],
+          message:
+            'Use typed hooks `useAppDispatch` and `useAppSelector` instead.',
+        },
+      ],
+      '@typescript-eslint/no-unused-vars': [
+        'error',
+        {
+          vars: 'all',
+          args: 'after-used',
+          destructuredArrayIgnorePattern: '^_',
+          ignoreRestSiblings: true,
+        },
+      ],
+      '@typescript-eslint/restrict-template-expressions': [
+        'warn',
+        {
+          allowNumber: true,
+        },
+      ],
+    },
+  },
+  {
+    files: ['**/__tests__/*.js', '**/__tests__/*.jsx'],
+
+    languageOptions: {
+      globals: {
+        ...globals.jest,
+      },
+    },
+  },
+  {
+    files: ['streaming/**/*.js'],
```

Comment:
I think this is the wrong approach. We should need an `eslint.config.mjs` file in `streaming/`. If we want to have some shared config between the two, we can import it from a common file.

But our frontend app and the streaming server are 2 very different apps, not using the same engine (web vs node), and have different requirements / rules / environment.

I remember you said in a previous PR that flat config means that all config needs to be in one directory, but this is not correct. Flat config means that this new format does not support as many ways of extending a config (or multi-level extends).

ESLint 9 will use the config file the closest to the file being linted, so if we have a config file in `streaming/`, it will be used for those files.

Reply:
Mmmh, that is weird, I remember having such a setup working. But maybe we ran `eslint` twice, with a different `cwd` everytime

Reply:
Got it working with the experimental flag after removing some of the `compat` stuff